plugins/terrain_and_color/main.py

```diff
@@ -588,3 +591,83 @@ def get_identify_detail(self, point):
         self.refresh()
 
         return None
+
+    def get_zonal_stats_from_point(self, point: Vector3) -> List[Optional[Dict]]:
+        results = []
+        if self.boundary_data:
+            for i, data_role in enumerate(self.data_roles):
+                dtype, _ = data_role
+                if dtype is DataPlugin.RASTER:
+                    plugin = self.get_data(i)
+                    if plugin is None:
+                        continue
+
+                    if i == 0:
+                        var = self._elevation_attribute.selected
+                    elif i == 1:
+                        var = self._attribute.selected
+                    else:
+                        var = ''
+
+                    raster = plugin.get_data(var)
+                    affine = plugin.affine
+                    res = plugin.resolution
+                    var_stats = plugin.variable_stats(var)
+                    nodata = var_stats.nodata_value
+
+                    # Transform point coordinates to crs of raster
+                    p = transform.xy(affine, point.x / res, point.y / res)
+                    zones = []
+                    for feat in self.boundary_data.get_features():
+                        if shapely.geometry.shape(feat['geometry']).contains(shapely.geometry.Point(*p)):
+                            zones.append(feat)
+
+                    # Retrieve zonal stats for this raster
+                    result = zonal_stats(zones, raster, affine=affine, nodata=nodata, add_stats=self.zonal_stats)
+                    for j, row in enumerate(result):
+                        row['Name'] = "{} (Zone {})".format(plugin.data_name, zones[j].get('id'))
+                    results.append(result)
+        return results
+
+    def get_zonal_stats_from_feature(self, feature: Dict) -> List[Optional[Dict]]:
```

Comment:
GeoJSON doesn't seem like the best in-memory representation to pass around geometries. Wouldn't shapely objects or such be more appropriate?

Reply:
Could be. I'll test and see if it is okay. Main reason for the GeoJSON interface is so that I could swap in the coordinates for the same features, but `rasterstats` might take the `shapely` objects just fine.

Reply:
Works fine.